### ISSUE TITLE
Non-conventional polynomials

### ISSUE THREAD

**Issue by rhaghi** [jonathf/chaospy#346]
Hi Jonathan,

Thanks a lot for this great package.

This is not really an issue/problem this is more a question. I want to use non-conventional polynomials for PCE. I know it is possible to define a "conventional" polynomial within Chaospy as you explained in the documentation. By non-conventional I mean polynomials that include sin(x), cos(x) or expontial components. An example of a non-conventional polynomial can be found in this [paper](https://doi.org/10.1002/we.2148). 

Looking forward to hearing your idea about that.

Cheers,
Rad

**Comment by jonathf** [comment]
Maybe.

Skimming the paper it looks like an application of intrusive Galerkin. Any chance you know of a paper that goes more into the methodology of this non-conventional approach?

I remember working on non-polynonial basis a bit during my doctorate, so I have an idea of what the topic is here, but think I need a refresher if I am to help you here.

**Comment by rhaghi** [comment]
Yes, it is intrusive Galerkin. What I want to do is using those polynomials in Chaospy for point collocation (chaospy.fit_regression). Unfortunately, I don't know any other example of that non-conventional approach in literature...

**Comment by jonathf** [comment]
Okay. Just so we are talking about the same thing here, can you describe what the orthogonality expansion will look like in the non-conventional approach?

The general orthogonality property being used in PCE is:
```
<p_n, p_m> = |p_n|^2 \delta_nm
```
where `\delta` is Kroenicker's delta. This shold to irrespectively of how the `p_n` looks like.

Conventionally `p_n` is defined as polynomials:
```
p_n = \sum_i c_ni q^i
```
and `c_ni` are coefficients chosen such that the orthogonality property over holds.

Would the non-conventional approach be the same, but have an expansion form that includes function transformations? Something like this for example:
```
p_n = \sum_i c_ni sin(q)^i
```

**Comment by jonathf** [comment]
I am sorry, I skimmed the paper to quickly. The form is listed in equation (19):
```
p_n = e^{2i c_n q}
```
or in the multivariate case:
```
p_n = \prod_d e^{2i c_nd q_d}
```
Yeah, I see how that would be a useful alternative projection approach.

Back to the question about how to use chaospy with this.
For the approach specifically in the paper I think the answer is no. For an approach to be compatible with chaospy you will need to be able to be a simple variable tranformation available that resulted in a polynomial. This assumes the form must conform to the format `p_n = \sum_i c_ni g(q)^i` for some simple function `g`. This is not the case here, unfortunatly.

I hope this makes sense, and sorry that chaospy can not provide more assistance.
